src/mlpack/core/optimizers/smorms3/smorms3.hpp

```diff
@@ -97,29 +91,12 @@ class SMORMS3
    * @param iterate Starting point (will be modified).
    * @return Objective value of the final point.
    */
+  template<typename DecomposableFunctionType>
```

Comment:
I know this is a bit tedious, but can you add the `@tparam DecomposableFunctionType ...` documentation to the `Optimize()` functions please since it's now removed from the class documentation?